### PR TITLE
On reload don't store duplicated keys

### PR DESCRIPTION
**Pull request by rohe**
If a key bundle is reloaded care should be taken to not store keys that already are in the key bundle.

- [X ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: tests/test_keyio.py [hunk 1]
```diff
@@ -575,3 +575,28 @@ def test_load_spomky_keys():
...
+    Emulates what happens if you fetch keys from a remote site and
+    you get back the same JWKS as the last time.
+    """
+    _jwks = {
+        "keys": [
+            {
+                "alg":"RS256",
+                "e":"AQAB",
+                "kty":"RSA",
+                "n":"wkpyitec6TgFC5G41RF6jBOZghGVyaHL79CzSjjS9VCkWjpGo2hajOsiJ1RnSoat9XDmQAqiqn18rWx4xa4ErdWVqug88pLxMVmnV9tF10uJNgIi_RSsIQz40J9aKrxOotN6Mnq454BpanAxbrbC5hLlp-PIGgmWzUDNwCSfnWBjd0yGwdYKVB6d-SGNfLvdMUhFiYIX0POUnJDNl_j3kLYQ0peYRbunyQzST5nLPOItePCuZ12G5e0Eo1meSF1Md3IkuY8paqKk-vsWrT22X7CUV3HZow06ogRcFMMzvooE7yDqS53I_onsUrqgQ2aUnoo8OaD0eLlEWdaTyeNAIw","use":"sig"
```
Comment (schlenk) [inline tests/test_keyio.py]: put the jwks in an extra file or make pylama happy about the overlong lines.

**Comment by rohe** [comment]
The order of the keys are irrelevant which make the idea about making it into a set pretty good :-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ The format is based on the [KeepAChangeLog] project.
 ## 0.14.0 [2018-05-15]
 
 ### Fixed
+- [#553] Made sure a a reload would not lead to duplicated keys in a keybundle.
```
[the remark this answers is not in the document]
Comment (rohe) [reply CHANGELOG.md]: Got it

**Comment by rohe** [comment]
Hmm, making _keys a set demands that jwkest.jwk.Key is hashable which is doable.
Could use thumbprint().

**Comment by rohe** [comment]
Let's do this first and then look at making _keys a set instead of a list.

**Pull request merged**